xcdat/axis.py

```diff
@@ -180,68 +222,174 @@ def swap_lon_axis(
 
+    # Attempt to swap the orientation for longitude coordinates.
     with xr.set_options(keep_attrs=True):
-        if to == (-180, 180):
-            new_lon = ((lon + 180) % 360) - 180
-            new_lon_bounds = ((lon_bounds + 180) % 360) - 180
-            ds = _reassign_lon(ds, new_lon, new_lon_bounds)
-        elif to == (0, 360):
-            new_lon = lon % 360
-            new_lon_bounds = lon_bounds % 360
-            ds = _reassign_lon(ds, new_lon, new_lon_bounds)
-
-            # Handle cases where a prime meridian cell exists, which can occur
-            # after swapping to (0, 360).
-            p_meridian_index = _get_prime_meridian_index(new_lon_bounds)
-            if p_meridian_index is not None:
-                ds = _align_lon_to_360(ds, p_meridian_index)
-        else:
-            raise ValueError(
-                "Currently, only (-180, 180) and (0, 360) are supported longitude axis "
-                "orientations."
-            )
+        for key in coord_keys:
+            new_coord = _swap_lon_axis(ds.coords[key], to)
+
+            if ds.coords[key].identical(new_coord):
+                continue
+
+            ds.coords[key] = new_coord
+
+        try:
+            bounds = ds.bounds.get_bounds("X")
+        except KeyError:
+            bounds = None
 
-    # If the swapped axis orientation is the same as the existing axis
-    # orientation, return the original Dataset.
-    if new_lon.identical(lon):
-        return dataset
+        if isinstance(bounds, xr.DataArray):
+            ds = _swap_lon_bounds(ds, str(bounds.name), to)
+        elif isinstance(bounds, xr.Dataset):
+            for key in bounds.data_vars.keys():
+                ds = _swap_lon_bounds(ds, str(key), to)
 
     if sort_ascending:
-        ds = ds.sortby(new_lon.name, ascending=True)
+        ds = ds.sortby(list(coord_keys), ascending=True)
 
     return ds
 
 
-def _reassign_lon(dataset: xr.Dataset, lon: xr.DataArray, lon_bounds: xr.DataArray):
+def _get_all_coord_keys(
+    obj: Union[xr.Dataset, xr.DataArray], axis: CFAxisKey
+) -> List[str]:
+    """Gets all dimension and non-dimension coordinate keys for an axis.
+
+    This function attempts to map an axis to its coordinate keys using the
+    CF axis and coordinate names with ``cf_xarray`` mapping tables.
+
+    Parameters
+    ----------
+    obj : Union[xr.Dataset, xr.DataArray]
+        The Dataset or DataArray object.
+    axis : CFAxisKey
+        The CF axis key ("X", "Y", "T", or "Z").
+
+    Returns
+    -------
+    List[str]
+        The axis coordinate variable keys.
     """
-    Reassign longitude coordinates and bounds to the Dataset after swapping the
-    orientation.
+    cf_attrs = CF_ATTR_MAP[axis]
+
+    keys: List[str] = []
+
+    try:
+        keys = keys + obj.cf.axes[cf_attrs["axis"]]
+    except KeyError:
+        pass
+
+    try:
+        keys = keys + obj.cf.coordinates[cf_attrs["coordinate"]]
+    except KeyError:
+        pass
+
```

Comment:
```suggestion

    if CF_ATTR_MAP['T']['coordinate'] in obj.coords.keys():
        keys = keys + [CF_ATTR_MAP[axis]['coordinate']]
        
```

This would fix [this issue](https://github.com/xCDAT/xcdat/pull/343#discussion_r1012275860). I think this logic is reasonable...